rubix-hadoop2/rubix-hadoop2CM/src/main/java/com/qubole/rubix/hadoop2/hadoop2CM/Hadoop2ClusterManager.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) 2016. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.hadoop2.hadoop2CM;
+
+import com.google.common.base.Throwables;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+import com.google.common.reflect.TypeToken;
+import com.qubole.rubix.spi.ClusterManager;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import com.qubole.rubix.spi.ClusterType;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+
+/**
+ * Created by sakshia on 28/7/16.
+ */
+public class Hadoop2ClusterManager
+        extends ClusterManager
+{
+    private boolean isMaster = true;
+    public int serverPort = 8088;
+    private String serverAddress = "localhost";
+    static LoadingCache<String, List<String>> nodesCache;
+    YarnConfiguration yconf;
+    String address = "localhost:8088";
+    private Log log = LogFactory.getLog(Hadoop2ClusterManager.class);
+    static String addressConf = "yarn.resourcemanager.webapp.address";
+    int refreshTime = super.getNodeRefreshTime();
```

Comment:
nit: no need of this, just use getNodeRefreshTime (without super) while creating nodeCache